File: src/Test/XAML/FeatureTests/Data/MarkupCompiler/ExtractedTestCases/TestCase2/vb/MappingPICode.vb

    Imports System
    Imports System.Windows
    Imports System.Windows.Navigation
    Imports System.Windows.Controls
    Imports Microsoft.Test

    <assembly:CLSCompliant(true)> _

    
    Namespace Local
        Public Class MyButton
            Inherits Button
        End Class
    End Namespace
    

    Namespace Harness
        Public Class OnStartupBVT
            Inherits System.Windows.Application

            Protected Overrides  Sub OnStartup(ByVal e As System.Windows.StartupEventArgs)
                Me.StartupUri = New Uri("MappingPIMarkup.xaml", UriKind.RelativeOrAbsolute)
                AddHandler Me.LoadCompleted, AddressOf OnLoad
                MyBase.OnStartup(e)
            End Sub
 
            Private  Sub OnLoad(ByVal sender As Object, ByVal e As NavigationEventArgs)
                Console.WriteLine("LoadCompleted event fired")
 
                ShutdownApp()
            End Sub
 
            Protected Overrides  Sub OnExit(ByVal e As ExitEventArgs)
                Console.WriteLine("Shutting down application")
            End Sub
 
            Private  Sub ShutdownApp()
                System.Windows.Application.Current.Shutdown(0)
                Microsoft.Test.Loaders.ApplicationMonitor.NotifyStopMonitoring()
            End Sub
        End Class

        Class EntryPoint
            <System.STAThread()> _ 
            Public Shared Function Main(ByVal args() As String) As Integer
                System.Threading.Thread.CurrentThread.SetApartmentState(System.Threading.ApartmentState.STA)
                Dim app As OnStartupBVT =  New OnStartupBVT() 
                Return app.Run()
            End Function
        End Class
           
    End Namespace  

File: src/Test/XAML/FeatureTests/Data/MarkupCompiler/ExtractedTestCases/TestCase1/cs/MappingPICode.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

using System;
    using System.Windows;
    using System.Windows.Navigation;
    using System.Windows.Controls;
    

    [assembly:CLSCompliant(true)]

    
    namespace Local
    {
        public class MyButton : Button
        {
        }
    }
    

    namespace Harness
    {
        public class OnStartupBVT : System.Windows.Application
        {
            protected override void OnStartup(System.Windows.StartupEventArgs e)
            {
                this.StartupUri = new Uri(@"MappingPIMarkup.xaml", UriKind.RelativeOrAbsolute);
                this.LoadCompleted += new System.Windows.Navigation.LoadCompletedEventHandler(OnLoad);
                base.OnStartup(e);
            }

            void OnLoad(object sender, NavigationEventArgs e)
            {
                Console.WriteLine("LoadCompleted event fired");

                ShutdownApp();
            }

            protected override void OnExit(ExitEventArgs e)
            {
                Console.WriteLine("Shutting down application");
            }

            private void ShutdownApp()
            {
                Application.Current.Shutdown(0);
                Microsoft.Test.Loaders.ApplicationMonitor.NotifyStopMonitoring();
            }
        }
           
        public class EntryPoint
        {
            [System.STAThread()]
            public static int Main(string[] args)
            {
                System.Threading.Thread.CurrentThread.SetApartmentState(System.Threading.ApartmentState.STA);
                OnStartupBVT app = new OnStartupBVT();
                return app.Run();
            }
        }
    }    

